stable/rabbitmq-ha/templates/statefulset.yaml

```diff
@@ -46,7 +46,11 @@ spec:
         - name: copy-rabbitmq-config
           image: {{ .Values.busyboxImage.repository}}:{{ .Values.busyboxImage.tag}}
           imagePullPolicy: {{ .Values.busyboxImage.pullPolicy }}
+          {{- if .Values.securityContext }}
+          command: ['sh', '-c', "cp /configmap/* /etc/rabbitmq; rm -f /var/lib/rabbitmq/.erlang.cookie; chown {{ .Values.securityContext.runAsUser }}:{{ .Values.securityContext.fsGroup }} /var/lib/rabbitmq"]
```

Comment:
Instead of duplicating, I prefer we turn this into a list with only the chown command having the if around it. Something like:

```yaml
        command:
        - "sh"
        - "-c"
        - |-
          cp /configmap/* /etc/rabbitmq
          rm -f /var/lib/rabbitmq/.erlang.cookie
          {{- if .Values.securityContext }}
          chown -R {{ .Values.securityContext.runAsUser }}:{{ .Values.securityContext.fsGroup }} /var/lib/rabbitmq
          {{- end }}
```

Reply:
Also, should be a recursive chown

Reply:
@steven-sheehy no problem I'll make those changes, however I am a little stumped - just now we had our `issue` again where (for a reason yet unknown) one of the pods restarts - this then leads to the pod throwing errors: `** Connection attempt from disallowed node` the cause of that error being `warning: /var/lib/rabbitmq/.erlang.cookie contents do not match RABBITMQ_ERLANG_COOKIE`

Until just now I didn't understand why this was being thrown - especially since we have persistent storage enabled which should mean that after the init container has first run the cookie exists and is set to the value provided by env var RABBITMQ_ERLANG_COOKIE so should always be the same. I then noticed that we are removing the `.erlang.cookie` file in the init container:
```
cp /configmap/* /etc/rabbitmq; rm -f /var/lib/rabbitmq/.erlang.cookie; chown 100:101 /var/lib/rabbitmq
```
so I can only assume that dockerentrypoint.sh is not writing or not able to write the correct env var back to that file? I guess one belt + braces way would be to compare the contents of the file to the env var and only update if they don't already match?

Reply:
Or your not supplying a static cookie to helm and thus to all your pods. By default the chart will randomly generate one, which changes every install.

Reply:
Yeah, I thought it might be that at first - but then I'd expect all nodes to have problems straight away since each node would generate it's own random cookie. Also after initially starting each pod exec'd into the pod and checked the value of on disk `/var/lib/rabbitmq/.erlang.cookie` against the env value of `RABBITMQ_ERLANG_COOKIE` and they matched

Reply:
You have to provide a static cookie, rabbitmq can't cluster without one. On startup, they all communicate over epmd and disallow nodes that don't match their own cookie.

Reply:
Yes, we are (we use helmfile) and set this:
```      
- rabbitmqErlangCookie: "{{ .Environment.Values.rabbitmq_erlang_cookie }}"
```
The value that comes from `"{{ .Environment.Values.rabbitmq_erlang_cookie }}"` - is the same on all cluster nodes/pods when they first start - it is _only_ after a crash/restart that this value is no longer being used